frontend/src/containers/Review/ReviewControl/ReviewControl.js

```diff
@@ -1,35 +1,93 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
 import { Form, Button } from "react-bootstrap";
-import {
-    PaperSpec,
-} from "../../../components";
+import { PaperSpec } from "../../../components";
+import { reviewActions, paperActions } from "../../../store/actions";
 import "./ReviewControl.css";
 
+
 class ReviewControl extends Component {
     constructor(props) {
         super(props);
         this.state = {
             title: "",
-            id: 5,
             content: "",
-            paper: {
-                id: 1,
-                title: "paper_title",
-                abstract: "abstract abstract abstract abstract abstract abstract abstract abstract ",
-                date: "2019.10.30.",
-                authors: "paper_authors",
-                keywords: [{ name: "paper_keywords", type: "author" }],
-                likeCount: 101,
-                reviewCount: 3,
-            },
+            paper: { },
+            likeCount: 0,
+            authorNames: [],
+            // keywords: [],
+            date: "",
+            thisReview: { id: 0 },
+            paperId: 0,
         };
         this.handleChange = this.handleChange.bind(this);
         this.clickButtonHandler = this.clickButtonHandler.bind(this);
     }
 
+    componentDidMount() {
+        if (this.props.mode === 0) {
+            this.props.onGetPaper({ id: this.props.match.params.paper_id })
+                .then(() => {
+                    this.setState({
+                        paper: this.props.selectedPaper,
+                        likeCount: this.props.selectedPaper.count.likes,
+                        authorNames: this.props.selectedPaper.authors.map((x) => `${x.first_name} ${x.last_name}`),
+                        date: this.props.selectedPaper.publication.date,
+                        title: "Enter title here.",
```

Comment:
I think it would be better that these kinds of words are placeholders of inputs, not real ready-input.

Reply:
Good!